haystack/components/evaluators/llm_evaluator.py

```diff
@@ -0,0 +1,230 @@
+        """
+        self.instructions = instructions
+        self.inputs = inputs
+        self.outputs = outputs or ["score"]
+        self.api = api
+        self.api_key = api_key
+        self.examples = examples
+        expected_inputs = dict(inputs)
+        if api == "openai":
+            self.generator = OpenAIGenerator(api_key=api_key)
+        else:
+            raise ValueError(f"Unsupported API: {api}")
+
+        component.set_input_types(self, **expected_inputs)
+
+    @component.output_types(results=List[List[Dict[str, Any]]])
+    def run(self, **inputs) -> Dict[str, Any]:
+        """
+        Run the LLM evaluator.
+
+        :param inputs:
+            The input values to evaluate. The keys are the input names and the values are lists of input values.
+        :returns:
+            A dictionary with a single `results` entry that contains a list of results.
+            Each result is a dictionary containing the keys as defined in the `outputs` parameter of the LLMEvaluator
+            and the evaluation results as the values.
+        """
+        self.validate_input_parameters(dict(self.inputs), inputs)
+        self.validate_lengths(*inputs.values())
+
+        template = self.prepare_template()
+        builder = PromptBuilder(template=template)
+
+        # inputs is a dictionary with keys being input names and values being a list of input values
+        # We need to iterate through the lists in parallel for all keys of the dictionary
+        input_names, values = inputs.keys(), list(zip(*inputs.values()))
+        list_of_input_names_to_values = [dict(zip(input_names, v)) for v in values]
+
+        results = []
+        for input_names_to_values in list_of_input_names_to_values:
+            prompt = builder.run(**input_names_to_values)
+            result = self.generator.run(prompt=prompt["prompt"])
+
+            self.validate_outputs(expected=self.outputs, received=result["replies"][0])
+            parsed_result = json.loads(result["replies"][0])
+            parsed_result["name"] = "llm"
+            results.append(parsed_result)
+
+        return {"results": results}
+
+    def prepare_template(self) -> str:
+        """
+        Combine instructions, inputs, outputs, and examples into one prompt template.
+
+        :returns:
+            The prompt template.
+        """
+        inputs_section = (
+            "{" + ",".join([f'"{input_socket[0]}": {{{{ {input_socket[0]} }}}}' for input_socket in self.inputs]) + "}"
+        )
+        examples_section = ""
+        if self.examples:
+            for example in self.examples:
+                examples_section += (
+                    "Inputs:\n" + json.dumps(example["inputs"]) + "\nOutputs:\n" + json.dumps(example["outputs"]) + "\n"
+                )
+        return (
+            f"Respond only in JSON format with a key {json.dumps(self.outputs)} and a value of either 0 for FALSE "
```

Comment:
Let's modify the template so that the final output looks something like this:
```
Instruction:
<User instruction>

Generate the response in JSON format with the following keys: 
<list of output keys>
Consider the instruction and the examples below to determine those values.

Examples:
<dump of the examples>
```
We shouldn't force the model to return boolean values for all keys - we might want to get explanations too, for instance. Hopefully, that's generic enough to get the job done, but we can iterate on the prompt when we implement the downstream metrics.